aten/src/ATen/native/TensorDimApply.h

```diff
@@ -0,0 +1,55 @@
+#include <ATen/ATen.h>
+
+namespace at {
+  namespace native {
+    //input tensors are non-zero dim and non-empty
+    template<typename T1, typename T2, typename Operation>
+    void tensor_dim_apply3(const Tensor& self, Tensor& values, Tensor& indices, int64_t dim, Operation op) {
+      int ndims = self.dim();
+      int tensor_dim_apply_has_finished = 0;
+      int tensor_dim_apply_i;
```

Comment:
don't declare tensor_dim_apply_i separately, declare it in the loop you are using it in